lib/fog/openstack/models/image_v2/images.rb

```diff
@@ -18,9 +18,9 @@ def summary(options = {})
           end
 
           def find_by_id(id)
-            image_hash = service.get_image_by_id(id).body
-            Fog::Image::OpenStack::V2::Image.new(
-                image_hash.merge(:service => service))
+            result = all.find { |image| image.id == id }
```

Comment:
This would not do an actual reload (new query), but take the cached value.

I suggest the following:

```ruby
def find_by_id(id)
  new(service.get_image_by_id(id).body)
end
```